Assets/MixedRealityToolkit.Staging/UnityAR/Editor/ConfigurationChecker/UnityARConfigurationChecker.cs

```diff
@@ -15,78 +17,108 @@ public class UnityARConfigurationChecker
-        private static FileInfo FindFile(string fileName)
+        /// <returns>True if the define was added, false otherwise.</returns>
+        private static bool EnsureArFoundationDefine()
         {
-            // Find the asmdef file
-            DirectoryInfo assets = new DirectoryInfo(Application.dataPath);
-            FileInfo[] files = assets.GetFiles(fileName, SearchOption.AllDirectories);
-            if (files.Length == 0) { return null; }
-            if (files.Length > 1)
+            const string fileName = "Unity.XR.ARFoundation.asmdef";
+            string[] definitions = { "ARFOUNDATION_PRESENT" };
+
+            FileInfo[] files = FileUtilities.FindFilesInPackageCache(fileName);
+            if (files.Length > 0)
             {
-                Debug.LogWarning($"More than one copy of {fileName} found in the project. Please ensure there are no duplicate resources and reload the project.");
-                return null;
+                ScriptingUtilities.AppendScriptingDefinitions(BuildTargetGroup.Android, definitions);
+                ScriptingUtilities.AppendScriptingDefinitions(BuildTargetGroup.iOS, definitions);
+                return true;
+            }
+            else
+            {
+                ScriptingUtilities.RemoveScriptingDefinitions(BuildTargetGroup.Android, definitions);
+                ScriptingUtilities.RemoveScriptingDefinitions(BuildTargetGroup.iOS, definitions);
+                return false;
             }
-
-            return files[0];
         }
 
         /// <summary>
-        /// Determines if a file update is needed and returns the replacement.
+        /// Updates the assembly definition to contain the appropriate references based on the Unity
+        /// version and if the project contains the AR Foundation package.
         /// </summary>
-        /// <param name="file">The reference file used to determine the new file.</param>
-        /// <returns>FileInfo object representing the file, or null if a new file is not needed.</returns>
-        private static FileInfo GetNewAsmDefFile(FileInfo file)
+        private static void UpdateAsmDef(bool arFoundationPresent)
         {
-            // Read the file.
-            string fileContents = string.Empty;
-            using (FileStream fs = new FileStream(file.FullName, FileMode.Open, FileAccess.Read))
+            string asmDefFileName = "Microsoft.MixedReality.Toolkit.Providers.UnityAR.asmdef";
+            FileInfo[] asmDefFiles = FileUtilities.FindFilesInAssets(asmDefFileName);
+
+            if (asmDefFiles.Length == 0)
             {
-                using (StreamReader reader = new StreamReader(fs))
-                {
-                    fileContents = reader.ReadToEnd();
-                }
+                Debug.LogWarning($"Unable to locate file: {asmDefFileName}");
+                return;
+            }
+
+            AssemblyDefinition asmDef = AssemblyDefinition.Load(asmDefFiles[0].FullName);
```

Comment:
>[0] [](start = 75, length = 3)

It seems odd for us to arbitrarily choose the first one - should we LogWarning if more than one are encountered?

Reply:
Added

```
Debug.LogWarning($"Mutliple ({asmDefFiles.Length}) {asmDefFileName} instances found. Modifying only the first.");
```